src/parser/vp9_parser.cpp

```diff
@@ -269,18 +390,51 @@ ParserResult Vp9VideoParser::FindFreeInDecBufPool() {
     }
     curr_pic_.dec_buf_idx = dec_buf_index;
     decode_buffer_pool_[dec_buf_index].use_status |= kFrameUsedForDecode;
-    // Todo decode_buffer_pool_[dec_buf_index].pic_order_cnt = curr_pic_.order_hint;
     decode_buffer_pool_[dec_buf_index].pts = curr_pts_;
     return PARSER_OK;
 }
 
 ParserResult Vp9VideoParser::FindFreeInDpbAndMark() {
     int i;
-    /*Todo*/
-
+    for (i = 0; i < VP9_NUM_REF_FRAMES; i++ ) {
+        if (dpb_buffer_.dec_ref_count[i] == 0) {
+            break;
+        }
+    }
+    if (i == VP9_NUM_REF_FRAMES) {
+        ERR("DPB buffer overflow!");
+        return PARSER_NOT_FOUND;
+    }
+    curr_pic_.pic_idx = i;
+    curr_pic_.use_status = kFrameUsedForDecode;
+    dpb_buffer_.frame_store[curr_pic_.pic_idx] = curr_pic_;
+    dpb_buffer_.dec_ref_count[curr_pic_.pic_idx]++;
+    // Mark as used in decode/display buffer pool
+    if (pfn_display_picture_cb_ && uncompressed_header_.show_frame) {
+        int disp_idx = curr_pic_.dec_buf_idx;
+        decode_buffer_pool_[disp_idx].use_status |= kFrameUsedForDisplay;
+        decode_buffer_pool_[disp_idx].pts = curr_pts_;
+        // Insert into output/display picture list
+        if (num_output_pics_ >= dec_buf_pool_size_) {
+            ERR("Display list size larger than decode buffer pool size!");
+            return PARSER_OUT_OF_RANGE;
+        } else {
```

Comment:
reverse the if else for better predicted code execution

Reply:
Done.